### DISCUSSION TITLE
Help setting up LTSP with LDAP-Authentification corectly

### DISCUSSION THREAD

**Discussion by emsfree** [ltsp/ltsp#604] [Q&A]
Hello there,
for several months now in our school we use an LTSP-environment (chrootles) for the teachers with a centralised LDAP authentification. The teachers all have one id for all the schools' services. Ever since, we seem to be unable to make the audio on the clients work. With local user login on the LTSP-Server, audio works perfectly, but when loging in via pxe from a machine, everything seems to work but audio.
Is there a guideline or "how to" achieve a fully working system? Do we need to use chroot or a vm as an image? Could that be the problem? I would very much like to understand how that works and then provide a proper description because I feel, that there is a need for such a setup, probably not only for us.
Thank you very much for ideas and help!
Regards,
Erasmus

**Comment by alkisg** [comment]
Hi, audio is supposed to work out of the box in LTSP setups, whether it's using chrootless, chroot or VM images.
LTSP doesn't do anything to handle audio, so it has no audio configuration options at all.
Everything is supposed to work the same way as in local installations, or on the server itself.

Which LTSP version are you using? What's the output of `ltsp --version`?
Does the audio work if you boot the clients with a local installation or a live USB?

**Comment by alkisg** [comment]
It shouldn't make any difference at all if the LTSP user is "local" or if it "comes from ldap", as far as audio is concerned.

The only thing that I can think of, is that you're somehow sending an older (or newer) kernel than the one in the image, so that the sound card modules do not exist in the image.

Check these in an LTSP client:

```shell
uname -r
du -sh /lib/modules/$(uname -r)
```

**Reply by emsfree** [reply]
HI, thank you for the reply!
Audio does work on the LTSP, as long as we log in with local users. Using LDAP, it doesn't anymore. So I guess our ldap-configuration is faulty, we missed some LTSP-specific settings for LDAP, probably.
The version-output is ltsp 21.01-1~ubuntu20.04.1.
alsamixer in a pxe-booted, ldap-authenticated client, returns to something like Error opening mixer-device. File or directory not found. No devices in the settings tab, no sound... It doesn't even find the local devices either, it seems.
Here is the rough setup of our system:
LTSP-setup
What are the essential steps to setup LTSP such that LDAP works flawlessly with this basic structure? Any special ltsp.conf settings?
Thanks for kind consideration. I am happy to provide a how-to as soon as we managed to get it work..

--

Wow, that was quick! - I screwed up editing my reply, though.
Thanks again!
In both cases the kernel is 5.11.0-38 generic. the `du...` returns `299M /lib/modules/5.11.0-38-generic`

**Comment by alkisg** [comment]
Oh I saw that you replied twice so I deleted one of them.
Then I guess you deleted the other one :D
I added your old text in your last reply so that there's some context.
Personally I prefer simple replies at the end of the page, not inline replies as they're very hard to read in a chronological order afterwards.

Regarding the issue, I can't imagine what else might be wrong; if you want, [join the chat room](https://ltsp.org/advanced/chat-room/) one of the following days for live troubleshooting.

**Comment by emsfree** [comment]
I'll do that, thank you!

Here is the rough setup of our system:

![LTSP-setup](https://user-images.githubusercontent.com/92694446/140656143-2dc5b687-9ed9-48d7-a9dd-b153bcc878a7.png)

Are there any specific configurations we need to follow in order to set this up fully working? I am more than happy to provide a How-to wikipage as soon as we got it running!
